File: Recommender/CommonRecommender/src/main/scala/com/lgr/recommender/MongoConfig.scala
package com.lgr.recommender

/**
 * @author Li Gengrun
 * @date 2023/3/23 16:12
 */

/**
 * mongoDB的配置信息
 * @param uri 连接
 * @param db  对应数据库
 */
case class MongoConfig(uri:String, db:String)

File: Backend/src/main/java/com/lgr/backend/config/MongoConfig.java
package com.lgr.backend.config;

import com.mongodb.MongoClient;
import com.mongodb.client.MongoDatabase;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * @author Li Gengrun
 * @date 2023/4/21 21:51
 */
@Configuration
public class MongoConfig {

    @Bean
    public MongoClient mongoClient() {
        return new MongoClient("localhost", 27017);
    }

    @Bean
    public MongoDatabase mongoDatabase() {
        return mongoClient().getDatabase("recommender");
    }
}

